src/ui/components/CustomButtons/CodeActionButton.jsx

```diff
@@ -6,14 +8,50 @@ import {
   CodeIcon,
   CopyIcon,
   TerminalIcon,
+  KeyIcon,
 } from '@primer/octicons-react';
-import React, { useState } from 'react';
+
+const API_BASE = import.meta.env.VITE_API_URI ?? '';
 
 const CodeActionButton = ({ cloneURL }) => {
   const [anchorEl, setAnchorEl] = useState(null);
   const [open, setOpen] = useState(false);
   const [placement, setPlacement] = useState();
   const [isCopied, setIsCopied] = useState(false);
+  const [protocol, setProtocol] = useState('https');
+
+  const [sshCfg, setSshCfg] = useState({ enabled: true, port: 22 });
+  useEffect(() => {
+    axios
+      .get(`${API_BASE}/api/v1/config/ssh`, { withCredentials: true })
+      .then((res) => {
+        const { enabled = true, port = 22 } = res.data || {};
```

Comment:
Is it necessary to add default values here if the API already sets the same default values? 🤔